File: KevinBankersen/ql/src/main/java/eu/bankersen/kevin/ql/ast/expr/logic/GT.java
package eu.bankersen.kevin.ql.ast.expr.logic;

import eu.bankersen.kevin.ql.ast.expr.Expr;
import eu.bankersen.kevin.ql.context.SymbolTable;
import eu.bankersen.kevin.ql.ast.expr.BooleanExpr;
import eu.bankersen.kevin.ql.ast.expr.EvaluateExeption;

public class GT extends BooleanExpr {

    public GT(final Expr lhs, final Expr rhs, final int line) {
	super(lhs, rhs, line);
    }

    @Override
    public final Boolean eval(SymbolTable symbolTable) throws EvaluateExeption {
	return (Integer) lhs().eval(symbolTable) > (Integer) rhs().eval(symbolTable);
    }
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/gt.go
package binaryoperatorexpr

import "ql/ast/expr"

type GT struct {
	BinaryOperator
}

func NewGT(lhs expr.Expr, rhs expr.Expr) GT {
	return GT{BinaryOperator{Lhs: lhs, Rhs: rhs}}
}

func (g GT) Eval(s interface{}) interface{} {
	return g.Lhs.Eval(s).(int) > g.Rhs.Eval(s).(int)
}
